braindecode/models/eegnet.py

```diff
@@ -50,27 +66,35 @@ class EEGNetv4(EEGModuleMixin, nn.Sequential):
 
     References
     ----------
-    .. [EEGNet4] Lawhern, V. J., Solon, A. J., Waytowich, N. R., Gordon,
-       S. M., Hung, C. P., & Lance, B. J. (2018).
-       EEGNet: A Compact Convolutional Network for EEG-based
-       Brain-Computer Interfaces.
-       arXiv preprint arXiv:1611.08024.
+    .. [EEGNet4] Lawhern, V. J., Solon, A. J., Waytowich, N. R., Gordon, S. M.,
+        Hung, C. P., & Lance, B. J. (2018). EEGNet: a compact convolutional
+        neural network for EEG-based brain–computer interfaces. Journal of
+        neural engineering, 15(5), 056013.
     """
 
     def __init__(
         self,
         n_chans=None,
         n_outputs=None,
         n_times=None,
-        final_conv_length="auto",
-        pool_mode="mean",
-        F1=8,
-        D=2,
-        F2=16,  # usually set to F1*D (?)
-        kernel_length=64,
-        third_kernel_size=(8, 4),
-        drop_prob=0.25,
+        final_conv_length: str | int = "auto",
+        pool_mode: str = "mean",
+        F1: int = 8,
+        D: int = 2,
+        F2: int = 16,
+        kernel_length: int = 64,
+        *,
+        depthwise_kernel_length: int = 16,
+        pool1_kernel_size: int = 4,
+        pool1_stride_size: int = 4,
+        pool2_kernel_size: int = 8,
+        pool2_stride_size: int = 8,
+        drop_prob: float = 0.25,
         activation: nn.Module = nn.ELU,
+        batch_norm_momentum: float = 0.01,
+        batch_norm_affine: bool = True,
+        batch_norm_eps: float = 1e-3,
+        conv_spatial_max_norm: int = 1,
         chs_info=None,
```

Comment:
For deprecation, add a `**kwargs` argument so it can still be passed and raise a warning if it is not empty?